src/patterns/ec2-app.ts

```diff
@@ -0,0 +1,77 @@
+import { HealthCheck } from "@aws-cdk/aws-autoscaling";
+import { Certificate } from "@aws-cdk/aws-certificatemanager";
+import { ApplicationProtocol, ListenerAction } from "@aws-cdk/aws-elasticloadbalancingv2";
+import { Duration } from "@aws-cdk/core";
+import { GuAutoScalingGroup, GuUserData } from "../constructs/autoscaling";
+import type { GuStack } from "../constructs/core";
+import { GuArnParameter } from "../constructs/core";
+import type { AppIdentity } from "../constructs/core/identity";
+import { GuVpc, SubnetType } from "../constructs/ec2";
+import { GuInstanceRole } from "../constructs/iam";
+import {
+  GuApplicationListener,
+  GuApplicationLoadBalancer,
+  GuApplicationTargetGroup,
+} from "../constructs/loadbalancing";
+
+interface GuEc2AppProps extends AppIdentity {
+  userData: GuUserData | string;
+  publicFacing: boolean; // could also name it `internetFacing` to match GuApplicationLoadBalancer
+  applicationPort: GuApplicationPorts | number;
+}
+
+export enum GuApplicationPorts {
```

Comment:
minor: I think this can be simplified a little into:
- `GuEc2App` which takes in a `port`
- `GuPlayEc2App` which extends `GuEc2App` and sets `port` to 9000
- `GuNodeEc2App` which extends `GuEc2App` and sets `port` to 3000

This would mean the `applicationPort` in `GuEc2AppProps` becomes a `number` type, which is a little simpler.

Reply:
This has been implemented in a new commit, good suggestion!